Microsoft.Azure.CosmosRepository.CleanArchitecture/IEtagMappedRepository.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Azure.CosmosRepository.CleanArchitecture;
+
+public interface IEtagMappedRepository<TItem, TMapped>
+    : IDisposable
+    where TItem : IItemWithEtag
+    where TMapped : class
+{
+    public string? Etag { get; }
+
+    public ValueTask<TMapped> GetAsync(
```

Comment:
We ought to consider reading collections and updating collections here

Reply:
I have started to update this, full support for all repository operations will be implemented.